exareme2/controller/services/flower/flower_algorithm_registry.py

```diff
@@ -0,0 +1,68 @@
+import asyncio
+from enum import Enum
+from enum import unique
+from logging import Logger
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from pydantic import BaseModel
+
+
+@unique
+class Status(str, Enum):
+    SUCCESS = "success"
+    RUNNING = "running"
+    FAILURE = "failure"
+
+
+class AlgorithmInputDataDTO(BaseModel):
+    data_model: str
+    datasets: List[str]
+    filters: Optional[dict]
+    y: Optional[List[str]]
+    x: Optional[List[str]]
+
+
+class FlowerAlgorithm(BaseModel):
+    inputdata: AlgorithmInputDataDTO = AlgorithmInputDataDTO(data_model="", datasets=[])
+    result: Dict = {}
+    status: Status = Status.SUCCESS
+
+
+class FlowerAlgorithmRegistry:
```

Comment:
Also the name of `FlowerAlgorithmRegistry` should probably change.

What about `FlowerExecutionInfo`?